### PR TITLE
Adding support for contentEncoding for draft7.

### PR DESCRIPTION
**Pull request by CheesyBoy123**
Motivation:

Adding support for contentEncoding which was added to the validation spec in draft7.
https://github.com/eclipse-vertx/vertx-json-schema/issues/139

Conformance:

You should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java [hunk 1]
```diff
@@ -928,12 +928,23 @@ private OutputUnit validate(final Object _instance, final JsonSchema schema, fin
...
-          schema.containsKey("format") &&
-            !Format.fastFormat(schema.get("format"), (String) instance)
-        ) {
+        if (schema.containsKey("format") &&
+            !Format.fastFormat(schema.get("format"), (String) instance)) {
           errors.add(new OutputUnit(instanceLocation, computeAbsoluteKeywordLocation(schema, schemaLocation + "/format"), baseLocation + "/format", "String does not match format \"" + schema.get("format") + "\"", OutputErrorType.INVALID_VALUE));
         }
+
+        //Content encoding was introduced in Draft7, but was turned into annotated only in draft 2019 and after.
+        if(draft.isAfter(Draft.DRAFT4) &&
```
Comment (pk-work) [inline src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java:937]: check formatting mo space after `if`

File: src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java [hunk 2]
```diff
@@ -928,12 +928,23 @@ private OutputUnit validate(final Object _instance, final JsonSchema schema, fin
         if (schema.containsKey("pattern") && !Pattern.compile(schema.get("pattern")).matcher((String) instance).find()) {
           errors.add(new OutputUnit(instanceLocation, computeAbsoluteKeywordLocation(schema, schemaLocation + "/pattern"), baseLocation + "/pattern", "String does not match pattern", OutputErrorType.INVALID_VALUE));
         }
-        if (
-          schema.containsKey("format") &&
-            !Format.fastFormat(schema.get("format"), (String) instance)
-        ) {
+        if (schema.containsKey("format") &&
```
Comment (pk-work) [inline src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java:931]: Isn't format only a annotation since 2020-12?

**Review by pk-work — CHANGES_REQUESTED** [review]

File: src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java (hunk 2 above)
Comment (CheesyBoy123) [reply src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java:931]: Yes but before we weren't checking the draft. I think in a separate change we can go through and fix some of these keywords that only exist in one draft or another

Comment (pk-work) [reply src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java:931]: Okay you are right, this PR is only about contentEncoding

**Review by pk-work — APPROVED** [review]

**Pull request merged**